docs/guide/sending-email.md

```diff
@@ -0,0 +1,226 @@
+      }
+    }
+  }
+};
+
+```
+<template v-slot:caption>
+/modules/@apostrophecms/email
+</template>
+
+</AposCodeBlock>
+
+## Using specific service APIs
+
+The `nodemailer` app has three built-in transports - `sendmail`, `SES` - for sending by AWS SES, and `stream` - which is used for testing. You can create your own custom transport using the `nodemailer` [transport API documentation](https://nodemailer.com/plugins/create/#transports), or use one of the many transport plugins that are available. These can either be found on the providers site, or by [searching NPM](https://www.npmjs.com/search?q=nodemailer%20transport). Each transporter will have slightly different parameters configured through an object that is then passed along with the specific transport.
+
+### Example for Mailgun
+
+<AposCodeBlock>
+
+```js
+const mg = require('nodemailer-mailgun-transport');
+
+const auth = {
+  auth: {
+    // key supplied with your Mailgun account
+    api_key: 'key-1234123412341234',
+    //one of your domains listed at your https://app.mailgun.com/app/sending/domains
+    domain: 'sandbox.domain.com'
+  }
+};
+
+module.exports = {
+  options: {
+    nodemailer: mg(auth);
+  }
+};
+
+```
+<template v-slot:caption>
+/modules/@apostrophecms/email
+</template>
+
+</AposCodeBlock>
+
+## Sending email from a module
+Once the `@apostrophecms/email` module is configured, email can be sent from any module using the `self.email(req, template, data, options)` method.
+
+The first parameter passed to this method is the page `req`.
+
+The next parameter, `template`, takes the name of a Nunjucks template that will make up the body of the email. This template should be located in the `views` template of the module. The method will pass this template, as well as an automatically generated version, to the `options` parameter of the `nodemailer` app.
+
+The `data` parameter takes an object that will be passed to the Nunjucks template for populating any customized fields. It can be accessed through `data.property` within the template.
+
+The final parameter, `options`, should be an object that contains the information for the email header. This is typically `from`, `to`, and `subject`. Any of these can have defaults set in the `options` of the `@apostrophecms/email` module, just like the transport. Any parameters specified in an individual module will override those set in this manner.
+
+### Example usage
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    lable: 'article',
+    pluralLabel: 'articles'
+  },
+  ...,
+  handlers(self) {
+    return {
+      'afterSave': {
+        async sendEmail(req, data) {
+          const options = {
+            from: 'admin@mysite.com',
+            to: 'editors@mysite.com',
+            subject: 'New Article added'
+          };
+          const template = 'email.html';
+          try {
+            await self.email(req, template, data, options);
+          return null;
```

Comment:
No need for any explicit return statement here.

Reply:
Removed.